microindex/findreader.go

```diff
@@ -0,0 +1,50 @@
+package microindex
+
+import (
+	"context"
+
+	"github.com/brimsec/zq/expr"
+	"github.com/brimsec/zq/pkg/iosrc"
+	"github.com/brimsec/zq/zbuf"
+	"github.com/brimsec/zq/zng"
+	"github.com/brimsec/zq/zng/resolver"
+)
+
+// FinderReader is zbuf.Reader version of Finder that streams back all records
+// in a microindex that match the provided key Record.
+type FinderReader struct {
+	compare expr.KeyCompareFn
+	finder  *Finder
+	reader  zbuf.Reader
+}
+
+func NewFinderReader(ctx context.Context, zctx *resolver.Context, uri iosrc.URI, inputs ...string) (*FinderReader, error) {
+	finder, err := NewFinder(ctx, zctx, uri)
+	if err != nil {
+		return nil, err
+	}
+	keys, err := finder.ParseKeys(inputs...)
```

Comment:
Do the error cases below here need to call `finder.Close()`?

Reply:
Good point!